docs/05_operations/08_update-queue/04_jms.mdx

```diff
@@ -0,0 +1,133 @@
+---
+title: 'Update queue - JMS'
+sidebar_label: 'JMS'
+id: jms
+keywords: [operations, update queue, jms, jakarta messaging]
+tags:
+- operations
+- update queue
+- jms
+- jakarta messaging
+---
+
+The Genesis low-code platform offers the flexibility to use a [Jakarta Messaging](https://en.wikipedia.org/wiki/Jakarta_Messaging) (former JMS) compliant message broker as its real-time update queue backend via [ArtemisMQ](https://activemq.apache.org/components/artemis/).
+
+Using a centralised external broker is highly recommended for complex, large or dynamically scaled clusters, as it reduces the complexity and overhead of peer-to-peer connectivity at the cost of maintaining or (re-using) a message broker.
+
+## Prerequisites
+
+Configuring Genesis to use Jakarta Messaging requires having **ArtemisMQ installed in the host** (or hosts in case of HA - see details [below](#high-avaiability)) as per [ArtemisMQ documentation](https://activemq.apache.org/components/artemis/documentation/).
+
+:::info
+Genesis doesn't ship Artemis MQ server, just its client library.
+:::
+
+## JMS configuration options
+
+### Basic configuration
+
+To use Jakarta Messaging in your Genesis application, you need to set the `MqLayer` in the system definition.
+
+```kotlin {title="genesis-system-definition.kts"}
+systemDefinition {
+    global {
+        ...
+        item(name = "MqLayer", value = "JMS")
+        ...
+    }
+}
+```
+
+All Available Config Variables:
+
+| Config Item                      | Description                          | Default                  |
+|----------------------------------|--------------------------------------|--------------------------|
+| `JmsHost` (required)             | The host name, can hold multiple values separated by comma, allowing HA failover | `http://localhost:61616` |
+| `JmsUsername`                    | Username to connect | |
+| `JmsPassword`                    | Password to connect (can be encrypted using `GenesisKey`) | |
+| `JmsTopicSubscriptionMode`       | Enum value (TRANSIENT, PERSISTENT) | `TRANSIENT`|
+| `JmsMaxConnectionRetryOnStartup` | The number of connections the process will attempt to make to the broker, before declaring the process `DOWN` | 20 |
+| `JmsReconnectDelayMs`            | The amount of time in milliseconds (exponentially) for a connection reattempt to be made, in case connectivity or fault happens | 1000 |
+| `JmsRetryToSendDelayMs`          | The amount of time in milliseconds for retrying sending a message in case of failure. | 1000 |
+
+### Durable consumers
+
+It is possible to configure certain process to receive messages that were sent when the process was offline. This might give another way for processes to recover from missing messages without the need of replaying an entire data stream.
```

Comment:
Benefit from re-wording here, not sure if it is entirely clear, looks too technical.